### DISCUSSION TITLE
How to use "SolveBiotSavart" module？

### DISCUSSION THREAD

**Discussion by zhifeiji-han** [SCIInstitute/SCIRun#2299] [General]
There is no  illustration about "SolveBiotSavart" module. 
What is the difference between "SolveBiotSavart" module and "SimulateForwardMagneticField" module？

**Comment by dcwhite** [comment]
Good question for @Sumientra

**Comment by moritzdannhauer** [comment]
Let me address your question.. Historically, SimulateForwardMagneticField has been around longer intended to be a multi-purpose module. Petar Petrov implemented the SolveBiotSavart module among other modules for simulating transcranial magnetic stimulation (TMS) in SCIRun4 and validated the results with measurements and published it:
https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0178952
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6024402/
His other modules allow you to design specific (TMS) coil geometries incompatible to work with SimulateForwardMagneticField and changing that would have made it difficult to maintain compatibility with already existing SCIRun4/5 networks/applications. 
We recognized the value of his implementation efforts and included his exact code into SCIRun5. 
I am sure you can contact him about any details. There are perhaps other differences that I am not thinking of right now as it has been a long time for me to use any of them.  I gave Sumientra a srn5 network for TMS simulations several years ago and you can ask her to provide you a recent SCIRun5 example.

**Comment by zhifeiji-han** [comment]
thanks you for your reply!
I wonder why there is only MEG(Magnetoencephalography), not MCG(MagnetoCardioGraphy) in SCIRun ?

**Comment by moritzdannhauer** [comment]
Not sure, and not my area but maybe you can modify (or get some helpful pointers for) an already existing cardiac-related network. In my mind it seems not to be fundamentally so different, just change the source model, model domain, and perhaps the sensor geometry to simulate MCG?
